tracing-core/src/metadata.rs

```diff
@@ -273,6 +273,30 @@ impl fmt::Display for Level {
     }
 }
 
+impl FromStr for Level {
+    type Err = ();
```

Comment:
I think this should return an actual error. Maybe something along the lines of
```rust
#[derive(Debug)]
pub struct ParseLevelError {
    _p: (),
}

impl fmt::Display for ParseLevelError {
    fn fmt(&self, f: &mut fmt::Formatter) -> fmt::Result {
        f.pad(
            "error parsing level: expected one of \"error\", \"warn\", \
             \"info\", \"debug\", \"trace\", or a number 1-5"
        )
    }
}

impl std::error::Error for ParseLevelError {}
```

Reply:
Thanks for your detailed feedback!

Reply:
Pleasure to help!